src/test/kotlin/uk/gov/gdx/datashare/service/DeathNotificationServiceTest.kt

```diff
@@ -2,78 +2,77 @@ package uk.gov.gdx.datashare.service
-import uk.gov.gdx.datashare.repository.ConsumerSubscriptionRepository
-import uk.gov.gdx.datashare.repository.EventData
-import uk.gov.gdx.datashare.repository.EventDataRepository
 import java.time.LocalDate
-import java.time.LocalDateTime
-import java.time.temporal.ChronoUnit
-import java.util.*
 
 class DeathNotificationServiceTest {
-  private val consumerSubscriptionRepository = mockk<ConsumerSubscriptionRepository>()
-  private val eventDataRepository = mockk<EventDataRepository>()
   private val levApiService = mockk<LevApiService>()
   private val objectMapper = JacksonConfiguration().objectMapper()
 
   private val underTest: DeathNotificationService = DeathNotificationService(
-    consumerSubscriptionRepository,
-    eventDataRepository,
     levApiService,
     objectMapper,
   )
 
-  private val ONE_SECOND_OFFSET = TemporalUnitWithinOffset(1, ChronoUnit.SECONDS)
-
   @Test
-  fun `mapDeathNotification maps string to full DeathNotificationDetails`() {
-    val input =
-      "{\"firstNames\":\"Alice\",\"lastName\":\"Smith\",\"dateOfBirth\":\"1910-01-01\",\"dateOfDeath\":\"2010-12-12\",\"address\":\"666 Inform House, 6 Inform street, Informington, Informshire\",\"sex\":\"female\"}"
-
-    val deathNotificationDetails = underTest.mapDeathNotification(input)
-
-    assertThat(deathNotificationDetails).isEqualTo(
-      DeathNotificationDetails(
-        firstNames = "Alice",
-        lastName = "Smith",
-        dateOfBirth = LocalDate.of(1910, 1, 1),
-        dateOfDeath = LocalDate.of(2010, 12, 12),
-        address = "666 Inform House, 6 Inform street, Informington, Informshire",
+  fun `getEnrichedData returns all data for a full set of enrichment fields`() {
+    val dataId = "123456789"
+    val datasetId = "DEATH_LEV"
+    val enrichmentFields = listOf(
+      "registrationDate",
+      "firstNames",
+      "lastName",
+      "sex",
+      "dateOfDeath",
+      "maidenName",
+      "dateOfBirth",
+      "address",
+      "birthPlace",
+      "deathPlace",
+      "occupation",
+      "retired",
+    )
+    val deathRecord = DeathRecord(
+      deceased = Deceased(
+        forenames = "Alice",
+        surname = "Smith",
+        dateOfBirth = LocalDate.of(1920, 1, 1),
+        dateOfDeath = LocalDate.of(2010, 1, 1),
         sex = "female",
+        address = "666 Inform House, 6 Inform street, Informington, Informshire",
+        birthplace = "Hospital",
+        deathplace = "Home",
+        maidenSurname = "Jones",
+        occupation = "Doctor",
+        retired = true,
       ),
+      id = dataId,
+      date = LocalDate.now(),
     )
-  }
 
-  @Test
-  fun `mapDeathNotification maps string to empty DeathNotificationDetails`() {
-    val input = "{}"
+    every { levApiService.findDeathById(dataId.toInt()) }.returns(listOf(deathRecord))
 
-    val deathNotificationDetails = underTest.mapDeathNotification(input)
+    val enrichedPayload = underTest.getEnrichedPayload(dataId, datasetId, enrichmentFields)
 
```

Comment:
Super nitpicky, but if we assert this isn't null can we drop all the question marks?